### PR TITLE
Update dev guide to add another option to run OpenSearch with Extensions feature-flag enable

### PR DESCRIPTION
**Pull request by DarshitChanpura**
### Description
Option 1, which is adding the feature flag in `gradle/run.gradle` did not work for me, but passing the feature flag as command line argument did. Hence, I'm adding alternate option, Option 2,  to run OpenSearch with Extensions feature-flag enabled in case someone runs into same issue as me.

### Issues Resolved
Related to #656 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by owaiskazi19** [comment]
@DarshitChanpura thanks for the PR. There's an option 3 as well.
You can uncomment the flag [here](https://github.com/opensearch-project/OpenSearch/blob/main/distribution/src/config/opensearch.yml#L116) and run `./bin/opensearch`. Probably you can add in this PR.

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -75,6 +76,20 @@ testClusters {
...
+- `./gradlew run -Dopensearch.experimental.feature.extensions.enabled=true` when running using gradle in developer mode
+
+##### Option 3
+Enable the experimental feature flag by setting it to true in `opensearch.yml`.
+- cd in to your local distribution build for OpenSearch.
+- now cd into config folder and open `opensearch.yml` in your local editor
+- Search for `opensearch.experimental.feature.extensions.enabled`, uncomment and set it to `true`
+- Run OpenSearch using 
+  - `./bin/opensearch` when running from a local distribution
+  - `./gradlew run` when running using gradle in developer mode
```
Comment (owaiskazi19) [inline DEVELOPER_GUIDE.md:91]: I don't think so we can run `./gradlew run` for this option. For gradle run we need to add the feature flag in run.gradle which is already covered in option 1

Comment (dbwiddis) [reply DEVELOPER_GUIDE.md:91]: I still haven't seen an example of the not-working run.gradle file.  That's how I'm running it and it works.

Comment (DarshitChanpura) [reply DEVELOPER_GUIDE.md:91]: true

Comment (DarshitChanpura) [reply DEVELOPER_GUIDE.md]: @dbwiddis I think I know where I got confused. After modifying run.gradle, I tried running the assembled version with `./bin/opensearch` as mentioned here thinking that the feature flag will be copied over. But that is not the case. run.gradle modification works only with `./gradlew run` and not `./bin/opensearch`

In order to run it with `./bin/opensearch` you need to either configure the feature flag via opensearch.yml or via command-line argument

**Review by owaiskazi19 — APPROVED** [review]

**Review by joshpalis — APPROVED** [review]

**Pull request merged**